src/main/java/xyz/parkh/doing/service/ScheduleService.java

```diff
@@ -0,0 +1,185 @@
+    // 1. 완료 O
+    // 2. 완료 X, 기한 O ( 기한 지난 것, 안 지난 것 구분)
+    // 3. 완료 X, 기한 X
+    public ResponseEntity<List<ScheduleShortInfo>> findScheduleList(
+            final String userIdInJwt, final String userId, final ScheduleStatusDto scheduleStatusDto) {
+        Boolean requestIsComplete = scheduleStatusDto.getIsComplete();
+        Boolean requestHasDeadLine = scheduleStatusDto.getHasDeadLine();
+
+        // 필수 인자가 없는 경우
+        if (userIdInJwt == null || userId == null || requestIsComplete == null) {
+            throw new ValueException("필수 인자가 없습니다.");
+        }
+
+        // 완료 되지 않은 일정 조회 시 기한이 있는 것 없는 것 구분 필수
+        if (!requestIsComplete && requestHasDeadLine == null) {
+            throw new ValueException("필수 인자가 없습니다.");
+        }
+
+        // TODO 친구 기능 추가 시 권한 확인하는 코드 작성 필요, 현재 본인 정보만 조회 가능.
+
+        // 본인이 아닐 경우
+        if (!userId.equals(userIdInJwt)) {
+            throw new DifferentAuthException("접근 할 수 있는 권한이 없습니다.");
+        }
+
+        List<ScheduleVo> scheduleList = scheduleMapper.selectByUserId(userId);
+
+        ArrayList<ScheduleShortInfo> scheduleShortInfoList = new ArrayList<>();
+
+        for (ScheduleVo schedule : scheduleList) {
+            Integer no = schedule.getNo();
+            String title = schedule.getTitle();
+            LocalDate endDate = schedule.getEndDate();
+            Boolean isComplete = schedule.getIsComplete();
+            LocalDate now = LocalDate.now();
+            Boolean overDeadLine = null;
+
+            // TODO 친구 기능 추가 시 조회한 schedule 의 isPublic 을 기반으로 공개 여부 추가 할 것.
+
+            // 완료된 일정 조회
+            if (requestIsComplete) {
+                if (!isComplete) continue; // 완료되지 않은 일정일 경우 처리하지 않음
+
+                // 기한이 있는 일정의 경우 기한 추가.
+                if (!Objects.isNull(endDate)) {
+                    overDeadLine = now.isAfter(endDate);
+                }
+
+                ScheduleShortInfo scheduleShortInfo = new ScheduleShortInfo().builder().no(no)
+                        .endDate(endDate).overDeadLine(overDeadLine).title(title).build();
+                scheduleShortInfoList.add(scheduleShortInfo);
+            } else if (requestHasDeadLine) {
+                // 완료 되지 않고 기한이 있는 일정 조회
+                if (isComplete || endDate == null) continue; // 완료 되었거나 기한이 없는 일정일 경우 처리하지 않음
+
+                overDeadLine = now.isAfter(endDate);
+
+                ScheduleShortInfo scheduleShortInfo = new ScheduleShortInfo().builder().no(no).title(title)
+                        .overDeadLine(overDeadLine).endDate(endDate).build();
+                scheduleShortInfoList.add(scheduleShortInfo);
+            } else {
+                // 완료 되지 않고 기한이 없는 일정 조회
+                if (isComplete || endDate != null) continue; // 완료 되었거나 기한이 있는 일정일 경우 처리하지 않음
+
+                ScheduleShortInfo scheduleShortInfo = new ScheduleShortInfo().builder().no(no).title(title).build();
+                scheduleShortInfoList.add(scheduleShortInfo);
+            }
+        }
```

Comment:
scheduleList 에서 각각의 요소를 가공하여 새로운 List 형태로 변환 시키는 로직으로 보이네요.
Stream 을 사용하면 리스트에서 stream 을 뽑아 map() 메소드를 이용하여 각 요소를 다른 객체로 쉽게 변환시킬 수 있습니다.

```suggestion
    scheduleList.stream()
        .map(schedule -> getScheduleShortInfo(schedule, requestIsComplete , requestHasDeadLine))
        .collect(Collectors.toList());
        
        
        public ScheduleShortInfo getScheduleShortInfo(ScheduleVo schedule, boolean requestIsComplete , boolean requestHasDeadLine){..........}
                }
```

Reply:
getScheduleShortInfo() 이 메소드에 분기 처리 로직을 넣어 조건에 따라 다른 ScheduleShortInfo 를 반환하도록 하면 라인 수를 줄일 수 있을 것 같습니다.
하나의 방법일 뿐이니 참고만 해주세요.